larch_feff/larch_feff.py

```diff
@@ -48,54 +48,44 @@ def main(structure_file: str, file_format: dict):
         if length != 1:
             raise ValueError(
-                f"Execpted single structure in cif file but found {length}"
+                f"Expected single structure in cif file but found {length}"
             )
+
+        # cif2feffinp below will take the absorber to be a chemical symbol, while this tool
+        # should support integer site index or chemical symbol of site (string). Hence convert
+        # any integer input to the relevant chemical symbol string.
+        # NOTE: The integer case for elements labeled with oxidation state 0 in the cif
+        # file, e.g. Pd0+.
         try:
-            aborsbing_atom = int(file_format["absorbing_atom"])
+            absorbing_atom = int(file_format["absorbing_atom"])
+            absorbing_atom = str(structures[0][absorbing_atom].specie)
         except ValueError:
-            # aborsbing_atom can be int or chemical symbol
-            aborsbing_atom = file_format["absorbing_atom"]
-
-        feff_header = Header(structures[0])
-        potential = Potential(structures[0], aborsbing_atom)
-        atoms = Atoms(structures[0], aborsbing_atom, file_format["radius"])
-        # if len(atoms.struct.sites) < len(potential.):
-
-        # print(atoms.as_dict())
-        feff_header.write_file(header_inp)
-        potential.write_file(potential_inp)
-        atoms.write_file(atoms_inp)
-        with open(path, "w") as feff_inp_file:
-            with open(header_inp) as f:
-                header_text = f.read()
-                print(header_text)
-                feff_inp_file.write(header_text + "\n")
-            with open(potential_inp) as f:
-                potential_text = f.readlines()
-                print(*potential_text)
-                feff_inp_file.writelines(potential_text + ["\n"])
-            with open(atoms_inp) as f:
-                atoms_text = f.readlines()
-                print(*atoms_text)
-                feff_inp_file.writelines(atoms_text + ["\n"])
-        if len(atoms_text) <= len(potential_text):
-            print(
-                "WARNING: Every potential in the structure must be represented"
-                " by at least one atom, consider increasing the radius"
-            )
+            absorbing_atom = file_format["absorbing_atom"]
+        
+        # NOTE: Here the first site listed in the cif file with the species 'absorbing_atom'
+        # is selected as the absorbing atom.
+        # NOTE: This generates output for FEFF6 via the 'version8' flag
+        inp_data = cif2feff.cif2feffinp(crystal_f, absorber=absorbing_atom, edge=None, 
+                                        cluster_size=file_format["radius"], absorber_site=1,
+                                        site_index=None, extra_titles=None, 
+                                        with_h=False, version8=False)
+        with open(path, 'w' ) as feff_inp_file:
+            feff_inp_file.write(inp_data+"\n")
+            print(inp_data+"\n")
+        
```

Comment:
We run `flake8` as part of the CI for the tools, which flagged a few changes:
https://github.com/MaterialsGalaxy/larch-tools/actions/runs/7933013896/job/21661007854?pr=38#step:6:15
These are just whitespace and line length so should be easy to change.